src/peft/utils/incremental_pca.py

```diff
@@ -66,11 +68,18 @@ def __init__(
             if lowrank_q < n_components:
                 raise ValueError("lowrank_q must be greater than or equal to n_components.")
 
-            def svd_fn(X):
-                U, S, V = torch.svd_lowrank(X, q=lowrank_q, niter=lowrank_niter)
+            def svd_fn(X, seed):
+                if seed is not None:
```

Comment:
I haven't used it, but could this be a good situation to use [`fork_rng`](https://pytorch.org/docs/stable/random.html#torch.random.fork_rng)?

Reply:
I didnt know about this function but it looks useful. I will check if I can implement it.